lib/Mojolicious/Command/Author/generate/app.pm

```diff
@@ -9,6 +9,10 @@ has usage       => sub { shift->extract_usage };
 sub run {
   my ($self, $class) = (shift, shift || 'MyApp');
 
+  unless($class =~ /^[a-zA-Z][a-zA-Z0-9_]*$/) {
```

Comment:
This can be one line with postfix unless. Please also use `\z` instead of `$` so that trailing newlines are not accepted.

Reply:
Double colons should also be allowed. Here is an example of a complete module name validation regex: https://metacpan.org/release/Module-Runtime/source/lib/Module/Runtime.pm#L176

Reply:
Done